### ISSUE TITLE
Adding server namespaces in final-server.json breaks redeployment

### ISSUE THREAD

**Issue by peetkes** [marklogic/marklogic-data-hub#1631]
If I add a namespace section to the final-server.json file `gradle mlRedeploy` runs successful. If I however add a namespace definition in it, it will break with ADMIN-DUPLICATECONFIGITEM
So this will work:
````
{
  "server-name": "%%mlFinalAppserverName%%",
  "server-type": "http",
  "root": "/",
  "group-name": "%%GROUP%%",
  "port": "%%mlFinalPort%%",
  "modules-database": "%%mlModulesDbName%%",
  "content-database": "%%mlFinalDbName%%",
  "authentication": "%%mlFinalAuth%%",
  "default-error-format": "json",
  "error-handler": "/MarkLogic/rest-api/error-handler.xqy",
  "url-rewriter": "/MarkLogic/rest-api/rewriter.xml",
  "rewrite-resolves-globally": true
  "namespace": []
}
````

But this will fail with beforementioned error:
````
{
  "server-name": "%%mlFinalAppserverName%%",
  "server-type": "http",
  "root": "/",
  "group-name": "%%GROUP%%",
  "port": "%%mlFinalPort%%",
  "modules-database": "%%mlModulesDbName%%",
  "content-database": "%%mlFinalDbName%%",
  "authentication": "%%mlFinalAuth%%",
  "default-error-format": "json",
  "error-handler": "/MarkLogic/rest-api/error-handler.xqy",
  "url-rewriter": "/MarkLogic/rest-api/rewriter.xml",
  "rewrite-resolves-globally": true
  "namespace": [
    "prefix": "someprefix",
    "namespace-uri": "some-namespace-uri"
  ]
}
````

The payload in the error message indeed shows the added namespace double!!!
I'm thinking that it is merging with itself.
In the DeployUserServersCommand.java class there is this function:
````
    protected JsonNode mergeServerFiles(File f) {
        List<File> files = new ArrayList<>();
        files.add(f);

        File userServerDir = hubConfig.getUserServersDir().toFile();
        File otherServerFile = new File(userServerDir, f.getName());
        if (otherServerFile != null && otherServerFile.exists()) {
            files.add(otherServerFile);
        }
        if (logger.isInfoEnabled()) {
            logger.debug("Merging JSON files at locations: " + files);
        }
        return JsonNodeUtil.mergeJsonFiles(files);
    }
````
Which given the final-server.json file from the ml-config/servers folder will be merged with itself...
I don't think this is supposed to happen.

This is also what you see in the logging info:
````
14:36:29.688 [DEBUG] [com.marklogic.hub.deploy.commands.DeployUserServersCommand] Merging JSON files at locations: [C:\Users\I25039\Projects\TFS\tradefinance-dh 
-v4\src\main\ml-config\servers\final-server.json, C:\Users\I25039\Projects\TFS\tradefinance-dh-v4\src\main\ml-config\servers\final-server.json] 
14:36:29.689 [INFO] [com.marklogic.mgmt.resource.appservers.ServerManager] Checking for existence of resource: tradefinance-dh4-FINAL 
14:36:29.689 [INFO] [com.marklogic.mgmt.ManageClient] Sending XML GET request as user 'admin' to path: /manage/v2/servers?group-id=Default 
14:36:29.689 [DEBUG] [org.springframework.web.client.RestTemplate] Created GET request for "https://localhost:8002/manage/v2/servers?group-id=Default" 
14:36:29.689 [DEBUG] [org.springframework.web.client.RestTemplate] Setting request Accept header to [text/plain, application/json, application/*+json, */*] 
14:36:29.689 [DEBUG] [org.apache.http.client.protocol.RequestAddCookies] CookieSpec selected: best-match 
14:36:29.689 [DEBUG] [org.apache.http.client.protocol.RequestAuthCache] Auth cache not set in the context 
14:36:29.689 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection request: [route: {s}->https://localhost:8002][total kept alive: 0 
; route allocated: 0 of 5; total allocated: 0 of 10] 
14:36:29.689 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection leased: [id: 3349][route: {s}->https://localhost:8002][total kept 
 alive: 0; route allocated: 1 of 5; total allocated: 1 of 10] 
14:36:29.689 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Opening connection {s}->https://localhost:8002 
14:36:29.690 [DEBUG] [org.apache.http.impl.conn.HttpClientConnectionOperator] Connecting to localhost/127.0.0.1:8002 
14:36:29.696 [DEBUG] [org.apache.http.impl.conn.HttpClientConnectionOperator] Connection established 127.0.0.1:63704<->127.0.0.1:8002 
14:36:29.697 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Executing request GET /manage/v2/servers?group-id=Default HTTP/1.1 
14:36:29.697 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Target auth state: UNCHALLENGED 
14:36:29.697 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Proxy auth state: UNCHALLENGED 
14:36:29.697 [DEBUG] [org.apache.http.impl.auth.HttpAuthenticator] Authentication required 
14:36:29.697 [DEBUG] [org.apache.http.impl.auth.HttpAuthenticator] localhost:8002 requested authentication 
14:36:29.697 [DEBUG] [org.apache.http.impl.client.TargetAuthenticationStrategy] Authentication schemes in the order of preference: [negotiate, Kerberos, NTLM, D 
igest, Basic] 
14:36:29.697 [DEBUG] [org.apache.http.impl.client.TargetAuthenticationStrategy] Challenge for negotiate authentication scheme not available 
14:36:29.697 [DEBUG] [org.apache.http.impl.client.TargetAuthenticationStrategy] Challenge for Kerberos authentication scheme not available 
14:36:29.698 [DEBUG] [org.apache.http.impl.client.TargetAuthenticationStrategy] Challenge for NTLM authentication scheme not available 
14:36:29.698 [DEBUG] [org.apache.http.impl.client.TargetAuthenticationStrategy] Challenge for Basic authentication scheme not available 
14:36:29.698 [DEBUG] [org.apache.http.impl.auth.HttpAuthenticator] Selected authentication options: [DIGEST [complete=true, nonce=null, nc=0]] 
14:36:29.698 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-3349: Close connection 
14:36:29.698 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Opening connection {s}->https://localhost:8002 
14:36:29.698 [DEBUG] [org.apache.http.impl.conn.HttpClientConnectionOperator] Connecting to localhost/127.0.0.1:8002 
14:36:29.705 [DEBUG] [org.apache.http.impl.conn.HttpClientConnectionOperator] Connection established 127.0.0.1:63705<->127.0.0.1:8002 
14:36:29.705 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Executing request GET /manage/v2/servers?group-id=Default HTTP/1.1 
14:36:29.705 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Target auth state: CHALLENGED 
14:36:29.705 [DEBUG] [org.apache.http.impl.auth.HttpAuthenticator] Generating response to an authentication challenge using digest scheme 
14:36:29.705 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Proxy auth state: UNCHALLENGED 
14:36:29.746 [DEBUG] [org.apache.http.impl.auth.HttpAuthenticator] Authentication succeeded 
14:36:29.747 [DEBUG] [org.apache.http.impl.client.TargetAuthenticationStrategy] Caching 'digest' auth scheme for https://localhost:8002 
14:36:29.747 [DEBUG] [org.springframework.web.client.RestTemplate] GET request for "https://localhost:8002/manage/v2/servers?group-id=Default" resulted in 200 ( 
OK) 
14:36:29.747 [DEBUG] [org.springframework.web.client.RestTemplate] Reading [java.lang.String] as "application/xml;charset=UTF-8" using [org.springframework.http 
.converter.StringHttpMessageConverter@6218a9a9] 
14:36:29.747 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-3349: Shutdown connection 
14:36:29.748 [DEBUG] [org.apache.http.impl.execchain.MainClientExec] Connection discarded 
14:36:29.748 [DEBUG] [org.apache.http.impl.conn.DefaultManagedHttpClientConnection] http-outgoing-3349: Close connection 
14:36:29.748 [DEBUG] [org.apache.http.impl.conn.PoolingHttpClientConnectionManager] Connection released: [id: 3349][route: {s}->https://localhost:8002][total ke 
pt alive: 0; route allocated: 0 of 5; total allocated: 0 of 10] 
14:36:29.749 [INFO] [com.marklogic.mgmt.resource.appservers.ServerManager] Found server with name of tradefinance-dh4-FINAL, so updating at path /manage/v2/serv 
ers/tradefinance-dh4-FINAL/properties?group-id=Default 
14:36:29.749 [INFO] [com.marklogic.mgmt.ManageClient] Sending JSON PUT request as user 'admin' to path: /manage/v2/servers/tradefinance-dh4-FINAL/properties?gro 
up-id=Default 
14:36:29.749 [DEBUG] [org.springframework.web.client.RestTemplate] Created PUT request for "https://localhost:8002/manage/v2/servers/tradefinance-dh4-FINAL/prop 
erties?group-id=Default" 
14:36:29.749 [DEBUG] [org.springframework.web.client.RestTemplate] Setting request Accept header to [text/plain, application/json, application/*+json, */*] 
14:36:29.749 [DEBUG] [org.springframework.web.client.RestTemplate] Writing [{"server-name":"tradefinance-dh4-FINAL","server-type":"http","root":"/","group-name" 
:"Default","port":"8111","modules-database":"tradefinance-dh4-MODULES","content-database":"tradefinance-dh4-FINAL","authentication":"digest","default-error-form 
at":"json","error-handler":"/MarkLogic/rest-api/error-handler.xqy","url-rewriter":"/MarkLogic/rest-api/rewriter.xml","rewrite-resolves-globally":true,"ssl-certi 
ficate-template":"tradefinance-local","ssl-require-client-certificate":"false","namespace":[{"prefix":"usr1","namespace-uri":"http://nl.abnamro.com/tradefinance 
-dh/user1"},{"prefix":"usr1","namespace-uri":"http://nl.abnamro.com/tradefinance-dh/user1"}]}] as "application/json;charset=UTF-8" using [org.springframework.ht 
tp.converter.StringHttpMessageConverter@6218a9a9] 
````

**Comment by peetkes** [comment]
I think this function should first check if it is merging with itself and if so, the mergeJsonFiles should also check the deeper levels.

**Comment by peetkes** [comment]
And this is basically all caused by the fact that the dhf does not properly handle namespaces. I used this on dhf 3 and this worked OK, but now when we try migrating to dhf4 we trip again on the fact that dhf generates invalid search options when using namespaces in the entities.

**Comment by akshaysonvane** [comment]
Hi, @peetkes this is a known issue. The fix for this has been added to the develop branch.

**Comment by aebadirad** [comment]
This should be fixed in 4.1.0 now, please reopen if it's not.
